src/Stateless/StateMachine.cs

```diff
@@ -494,7 +497,7 @@ public bool IsInState(TState state)
         /// <returns>True if the trigger can be fired, false otherwise.</returns>
         public bool CanFire(TTrigger trigger)
         {
-            return CurrentRepresentation.CanHandle(trigger);
+            return CurrentRepresentation.CanHandle(trigger, new object[_triggerConfiguration.TryGetValue(trigger, out var triggerConfiguration) ? triggerConfiguration.Arity : 0]);
```

Comment:
Any way to avoid the extra allocation here? (Declaring a static array of stand-in argument arrays, and then indexing into it by `Arity`, could do it - but with that level of machinery then perhaps `TriggerWithParameters` could just initialize and store a surrogate argument array and provide that directly instead of `Arity`?)

Reply:
@nblumhardt I considered that, but this actually does less work because, in practice, `Arity` is overridden by all of its implementors to return a constant value and the reflection based method would only ever be called if someone were to add additional implementations in a client library.  The reason this seemed better than just leaving it abstract is that `Arity` can remain internal but still be compatible with a client library that follows the same pattern/conventions but used more parameters than are provided in Stateless.

That being said, I don't think the difference is that significant so it can certainly be changed as you suggest.  Do you still think adding a static array is the way to go here?